azurelinuxagent/ga/exthandlers.py

```diff
@@ -1091,28 +1097,61 @@ def get_status_file_path(self, extension=None):
-            ext_status.status = "error"
+            msg = u"Failed to find any status for extension - {0}-{1}, Sequence number {2}. Failed due to {3}" \
+                .format(ext.name, self.ext_handler.properties.version, seq_no, e)
+            ext_status.message = msg
+            ext_status.code = ExtensionErrorCodes.PluginUnknownFailure
+            ext_status.status = EXTENSION_STATUS_WARNING
         except ValueError as e:
-            ext_status.message = u"Malformed status file {0}".format(e)
-            ext_status.code = -1
-            ext_status.status = "error"
+            msg = u"Failed to read any status for extension - {0}-{1}, Sequence number {2}. Failed due to {3}"\
+                .format(ext.name, self.ext_handler.properties.version, seq_no, e)
+            ext_status.message = msg
+            ext_status.code = ExtensionErrorCodes.PluginUnknownFailure
+            ext_status.status = EXTENSION_STATUS_WARNING
+
+        if ext_status.code == ExtensionErrorCodes.PluginUnknownFailure:
+            logger.periodic_warn(logger.EVERY_HALF_HOUR, ext_status.message)
+            add_periodic(delta=logger.EVERY_HALF_HOUR, name=ext.name, version=self.ext_handler.properties.version,
+                         op=WALAEventOperation.StatusProcessing, is_success=False, message=ext_status.message,
+                         log_event=False)
+        else:
+            # We did not encounter any PluginUnknownFailures and thus the status file was correctly written and has
+            # valid json.
+            try:
+                parse_ext_status(ext_status, data)
+                if len(data_str) > MAX_STATUS_FILE_SIZE_IN_BYTES:
+                    raise ExtensionStatusError("For Extension Handler {0}-{1}, the status file {2} of size {3} bytes is"
+                                               " too big. Max Limit allowed is {4} bytes"
+                                               .format(ext.name, self.ext_handler.properties.version, ext_status_file,
+                                                       len(data_str), MAX_STATUS_FILE_SIZE_IN_BYTES),
+                                               code=ExtensionStatusError.MaxSizeExceeded)
+            except ExtensionStatusError as e:
+                logger.periodic_warn(logger.EVERY_HALF_HOUR, ustr(e))
+                add_periodic(delta=logger.EVERY_HALF_HOUR, name=ext.name, version=self.ext_handler.properties.version,
+                             op=WALAEventOperation.StatusProcessing, is_success=False, message=ustr(e), log_event=False)
+
+                if e.code == ExtensionStatusError.MaxSizeExceeded:
+                    # Emptying the substatus to reduce the size, and preserve other fields of the text
+                    ext_status.substatusList = []
+                    ext_status.message = ext_status.message if len(ext_status.message) < 200 \
+                        else ext_status.message[:200] + " ... TRUNCATED MESSAGE"
```

Comment:
Suggestion - `... TRUNCATED MESSAGE` seems a bit verbose, I think the same effect could be achieved more concisely with `[TRUNCATED]` at the end of the message.

Reply:
This looks cleaner.